ledger/complete/mtrie/trie/trie_test.go

```diff
@@ -89,7 +89,7 @@ func Test_TrieWithManyRegisters(t *testing.T) {
 	paths, payloads := deduplicateWrites(sampleRandomRegisterWrites(rng, 12001))
 	updatedTrie, err := trie.NewTrieWithUpdatedRegisters(emptyTrie, paths, payloads)
 	require.NoError(t, err)
-	expectedRootHashHex := "18a7c33a0ecf148274f860246f23dffdc6d15dc846e0ae34f6887a43ec67124c"
+	expectedRootHashHex := "58042aca145b316263581d1789a1fc50ac2844f1df08cb006d0849e788f6b754"
```

Comment:
Please note that this test has lost a lot of its power:
* the `LinearCongruentialGenerator` only generates 16bit outputs. Therefore, we are only testing with compactified leafs. 
* this can be avoided by padding the output with 0 at the front as opposed to the end (i.e. by updating `utils.PathByUint16`)


May I request to add the following goDoc to https://github.com/onflow/flow-go/blob/3b8b3e3b8e96028f0ff414bd6d92b19b109a5c6f/ledger/complete/mtrie/trie/trie_test.go#L200
```
    simple Linear congruential RNG
    https://en.wikipedia.org/wiki/Linear_congruential_generator
    with configuration for 16bit output used by Microsoft Visual Basic 6 and earlier
```

Reply:
added comment